### PR TITLE
create a HTTPclient method to modify the requests timeout

### PR DESCRIPTION
**Pull request by lucasmenendez**

### CODE REVIEW

**Review by altergui — APPROVED** [review]
LGTM

File: apiclient/client.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ const (
 	// DefaultRetries this enables Request() to handle the situation where the server replies
 	// "mempool is full", it will wait for next block and retry sending the tx
 	DefaultRetries = 3
+	// DefaultTimeout is the default timeout for the HTTP client
+	DefaultTimeout = 10 * time.Second
```
Comment (altergui) [inline apiclient/client.go:36]: where is this used?

Comment (lucasmenendez) [reply apiclient/client.go:36]: ops, missing commit :P

File: apiclient/client.go [hunk 2]
```diff
@@ -144,6 +146,11 @@ func (c *HTTPclient) SetRetries(n int) {
 	c.retries = n
 }
 
+func (c *HTTPclient) SetTimeout(d time.Duration) {
+	c.c.Timeout = d
+	c.c.Transport.(*http.Transport).ResponseHeaderTimeout = d
```
Comment (altergui) [inline apiclient/client.go]: i have the feeling this can panic. better make an assertion and simply skip if it's not http.Transport

Comment (altergui) [reply apiclient/client.go]:
```go
func (c *HTTPclient) SetTimeout(d time.Duration) {
	c.c.Timeout = d
	if transport, ok := c.c.Transport.(*http.Transport); ok {
		transport.ResponseHeaderTimeout = d
	}
}

```

Comment (altergui) [reply apiclient/client.go]: i mean in our codebase it's always http.Transport. but it also doesn't hurt to avoid panics

**Pull request merged**